accepted/00000-maintenance-windows.md

```diff
@@ -0,0 +1,142 @@
+
+# Architecture
+[architecture]: #architecture
+
+A system is in "Maintenance Mode" or not.
+When a system is in "Maintenance Mode" no action can be executed.
+Setting the Maintenance Mode "on" and "off" requires Organization Admin permissions.
+A "Maintenance Window" does nothing else than turning Maintenance Mode "on" when the window starts
+and "off" when it ends.
+
+Defining Maintenance Windows requires Organization Admin permissions.
+
+Maintenance Windows are defined on system level.
+Every System can get 1 Schedule assigned.
+A Schedule can be used for multiple systems.
+Inside of the Schedule, multiple Maintenance Windows can be defined.
+They can be defined as individual or as recurring events.
+
+To assign a Schedule to multiple systems, SSM should be used.
+
+
+# Detailed design
+[design]: #detailed-design
+
+## Implement a Maintenance Mode
+
+Check, how System Locking is implemented. Copy the mechanism for Maintenance Mode.
+Does it make sense to use System Locking as "Maintenance Mode"?
+
+For the first iteration we want to block all actions when the system is not in maintenance mode.
+In future iterations we can consider to allow some actions which are mostly read-only (package list refresh or hardware refresh).
+Also actions which prepare things for the maintenance window could be considered (pre-fetching packages for update).
+
+
+## Maintenance Schedule
+
+A Maintenance Schedule is a set of Maintenance Windows. A Maintenance Schedule has a name/label.
+The data should be stored as ICal data (https://tools.ietf.org/html/rfc5545).
+Importing data should be possible. (URL for external download?)
+
+We should use a java lib for ical like ical4j(https://github.com/ical4j/ical4j) to parse
+the ics files.
+Every `VEVENT` describe a maintenance window (or should we require a special `SUMMARY` value?).
+Events for a full day (anniversary) or multiple days should also be considered.
+Recurring events must be considered as well.
+
+We need to implement an overview page listing all schedules.
+We need the possibilities to add/edit/remove schedules.
+
+Watching details of a schedule could be done in a calendar widget.
+
+
+## Maintenance Window
+
+A Maintenance Window specify a time when changes on a system can be applied.
+It has a start date-time and an end date-time.
+It can be defined as recurring interval.
+
+When a Maintenance Window start, it turn the maintenance mode "on".
+When its end, it turns the maintenance mode "off".
+
+
+## Assign a Maintenance Schedule to a System
+
+There could be multiple ways to assign systems to maintenance schedules.
+
+- In the systems details page we need to show the selected schedule with a view and a change option.
+- It should be possible to change the maintenance schedule for multiple systems using SSM
+- In the details of the Maintenance Schedule we could have a systems page where add/remove is possible.
```

Comment:
As I understood it changing the schedule would change the schedule for all systems that follow that schedule.